orderapi/src/main/java/fi/hel/verkkokauppa/order/service/elasticsearch/SearchAfterService.java

```diff
@@ -0,0 +1,112 @@
+package fi.hel.verkkokauppa.order.service.elasticsearch;
+
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.*;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.SortBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.data.elasticsearch.core.query.NativeSearchQuery;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+@Service
+@Slf4j
+public class SearchAfterService {
+
+    @Autowired
+    private RestHighLevelClient highLevelClient;
+
+    @Autowired
+    private Environment env;
+
+    /**
+     * Build Json query based on NativeSearchQueryBuilder with added search after parameters.
+     *
+     * For first pass businessSortMarker and tiebreakerSortMarker can be null,
+     * for search after they are needed.
+     */
+
+    public SearchRequest buildSearchAfterSearchRequest(NativeSearchQuery query,
+                                                       SortBuilder[] sorts,
+                                                       String... indices) throws Exception {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+
+        if( indices == null || indices.length == 0 )
+        {
+            // throw exception, indice is needed
+            throw new Exception("buildSearchAfterSearchRequest: at least one indice is needed");
+        }
+
+        if( query != null && !query.toString().isEmpty() )
+        {
+            // set query
+            searchSourceBuilder.query(query.getQuery());
+        }
+
+        if( sorts != null )
+        {
+            // set sorts
+            for (SortBuilder sort : sorts) {
+                searchSourceBuilder.sort(sort);
+            }
+        }
+
+        // set number of records retrieved with one query
+        searchSourceBuilder.size(Integer.parseInt(env.getProperty("elasticsearch.search-after-page-size")));

```

Comment:
@Value("${elasticsearch.search-after-page-size}")
private int elasticsearchSearchAfterPageSize;

Reply:
Change to this